fuzzit.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -xe
+
+# Go-fuzz doesn't support modules yet, so ensure we do everything in the old style GOPATH way
+export GO111MODULE="off"
+
+# Install go-fuzz
+go get -u github.com/dvyukov/go-fuzz/go-fuzz github.com/dvyukov/go-fuzz/go-fuzz-build
+
+# Compiling prometheus fuzz targets in fuzz.go with go-fuzz (https://github.com/dvyukov/go-fuzz) and libFuzzer support
+go get -u ./protoc-gen-grpc-gateway/httprule
+go-fuzz-build -libfuzzer -o parse-http-rule.a ./protoc-gen-grpc-gateway/httprule
+clang-9 -fsanitize=fuzzer parse-http-rule.a -o parse-http-rule
+
+wget -q -O fuzzit https://github.com/fuzzitdev/fuzzit/releases/download/v2.4.29/fuzzit_Linux_x86_64
+chmod a+x fuzzit
+
+if [ -z "CIRCLE_PULL_REQUEST" ]; then
```

Comment:
Is this going to be set on the master branch?

Reply:
If it's a Pull-Request I assume yes. the idea is to run fuzzing only when new code is pushed to master and regression otherwise, i.e other branches and PRs.
Maybe we can just set a nightly that will run fuzzing jobs and otherwise we will run regression. (I hope it wasn't to confusing)

Reply:
Depending on what you're trying to accomplish here, maybe we should just use the `CIRCLECI` environment variable instead to indicate that we're running in Circle? See https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables.

![image](https://user-images.githubusercontent.com/6604151/63415302-70e57300-c3f5-11e9-9511-778a08c6bb65.png)

Reply:
@johanbrandhorst The intention is to run regression tests¹ when CI runs on a pull request, and to upload a new fuzzer for continuous testing when code has been pushed to master².
CIRCLE_PULL_REQUEST will be set for PRs, but (presumably) not for pushes to master.

¹: Regression test in this case means that the fuzzer binary runs on an already generated corpus of inputs, which were generated during normal fuzzing. Think of it as additional tests to run. The fuzzit CLI calls this "local-regression" because the job runs locally, i.e. in the CircleCI container, as opposed to uploading the fuzzer to fuzzit.dev.

²: This of course assumes that CircleCI was configured to also triggers for pushes to master, not just pull requests.

Reply:
Yes I can add this check but CIRCLE_PULL_REQUEST will still be needed

Reply:
Nevermind, @dominikh answered my concerns. Please don't change it.